p2p/host/basic/basic_host.go

```diff
@@ -96,12 +97,14 @@ type BasicHost struct {
 
 	addrChangeChan chan struct{}
 
-	addrMu                 sync.RWMutex
+	addrMu                 sync.RWMutex // protects the following fields
 	updateLocalIPv4Backoff backoff.ExpBackoff
 	updateLocalIPv6Backoff backoff.ExpBackoff
 	filteredInterfaceAddrs []ma.Multiaddr
 	allInterfaceAddrs      []ma.Multiaddr
 
+	relayAddrs atomic.Value
```

Comment:
Slight preference to using the atomic.Pointer type since it's a little nicer to use.